pkg/controller/certificates/readiness/readiness_controller.go

```diff
@@ -175,12 +176,14 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		crt.Status.NotBefore = nil
 		crt.Status.RenewalTime = nil
 	}
-	_, err = c.client.CertmanagerV1().Certificates(crt.Namespace).UpdateStatus(ctx, crt, metav1.UpdateOptions{})
-	if err != nil {
-		return err
+	if reflect.DeepEqual(oldCrt.Status, crt.Status) == false {
```

Comment:
As @irbekrm noted, I think we should use the semantic equality function here instead: https://github.com/jetstack/cert-manager/issues/3663#issuecomment-800538995

If we do not, things like timestamps that are identical can sometimes register as different, as the `reflect` package performs a comparison by comparing struct fields (and there is a `time.Time` embedded within `metav1.Time`, which can get a bit weird with comparisons due to some of the internal timezone related fields)

Reply:
Also, a bit of a nit, but can we change this to `if !apiequality.Semantic.DeepEqual(...) {` instead of using `== false`? 😄

Reply:
Made the [changes](https://github.com/jetstack/cert-manager/blob/086648fea97e16e3f0f370b8aaf0d1067a28a35c/pkg/controller/certificates/readiness/readiness_controller.go#L179), thanks. :smile: